run-custom-mao.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash -xe
+
+# This script needs an machine-api-operator image as an input parameter
+# For example: ./run-custom-mao.sh quay.io/sdasu/machine-api-operator:sdasu-fix
+export SCRIPTS_DIR=$(pwd)
+export CUSTOM_MAO_BUILD=$1
+
+kubectl get secrets --namespace openshift-machine-api | grep mariadb
+
+# Stop CVO, MAO, machine-api-controller and Metal3 deployments
+oc --config /home/stack/dev-scripts/ocp/auth/kubeconfig scale deployment --replicas=0 cluster-version-operator --namespace openshift-cluster-version
+oc --config /home/stack/dev-scripts/ocp/auth/kubeconfig scale deployment -n openshift-machine-api --replicas=0 machine-api-operator
+oc --config /home/stack/dev-scripts/ocp/auth/kubeconfig scale deployment --replicas=0 machine-api-controllers --namespace openshift-machine-api
+oc --config /home/stack/dev-scripts/ocp/auth/kubeconfig scale deployment --replicas=0 metal3 --namespace openshift-machine-api
+
+REPLACEMENT_TEXT="image: ${CUSTOM_MAO_BUILD}"
+echo $REPLACEMENT_TEXT
+
+oc --config /home/stack/dev-scripts/ocp/auth/kubeconfig get deployments -n openshift-machine-api machine-api-operator -o yaml > mao.yaml
+sed -i -e "s!image: .*!${REPLACEMENT_TEXT}!g" mao.yaml
+oc --config /home/stack/dev-scripts/ocp/auth/kubeconfig apply -f mao.yaml
+
+# Give the machine-api-operator rbac access to create and read secrets
+oc --config /home/stack/dev-scripts/ocp/auth/kubeconfig apply -f machine-api-operator-roles.yaml
+
+oc --config /home/stack/dev-scripts/ocp/auth/kubeconfig scale deployment -n openshift-machine-api --replicas=1 machine-api-operator
```

Comment:
I don't think you need to scale down/up the MAO.  Once you stop CVO and patch the MAO deployment, it's going to restart automatically.

Reply:
Yes. I was thinking about that. By forcing a scale up right away we can save some time (time for the reconcile method to run again and discover that the MAO image has changed and then bring it down and up again with the new image) during our CI runs. I am open to removing this and let the reconcile do its thing.

Reply:
I tried this yesterday. "oc patch" can be used to update a pod's image and that would automatically restart the pod with the image. The only issue I had with using "oc patch" was that I could not get it to work using the image name to be replaced as a variable. Also, that still needed the same number of steps as updating the deployment object as I have now. (1. oc get pods 2. Grab the machine-api-operator pod name 3. Update the image name within the pod)
So, continuing to use the existing approach to update the deployment. When we update the deployment object, then we specifically need to scale machine-api-operater down and up for it to pick up the new image.